kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -358,7 +357,7 @@ public Long getOffsetToCommitAndReset() {
   }
 
   public Long currentStartOffset() {
-    return currentStartOffset;
+    return minStartOffset();
```

Comment:
this feels like doing two fixes in the same PR. should this fix and its tests go to a separate PR?

Reply:
Please see my response to @raphaelauv above: https://github.com/confluentinc/kafka-connect-storage-cloud/pull/461#issuecomment-989554722 He had the same concern as you.

Shortly, without other changes from this PR, there are no problems with this method, so no need to fix it separately.

This change is covered by integration tests added in this PR. Do you want me to also add a unit test for this?

Reply:
Thanks @lion-simba for the explanation, it now makes sense to keep in this PR. 

If you can add a test it would be great :).

Reply:
I was about to add a unit test, but suddenly realized that tests defined in `io.confluent.connect.s3.TopicPartitionWriterTest` are actually kind of **integration tests** for TopicPartitionWriter: they communicate with mocked S3 instance, which is similar to how my tests in this PR work. So, if I add a new test for this particular code change to TopicPartitionWriterTest in a similar way, then this new test will be pretty much a duplicate to my tests added in this PR. And it makes a little sense to me to do so.

It looks like there is no other test types for TopicPartitionWriter exist. So, maybe it is okay to have only tests added in this PR? What do you think?

Reply:
Thanks, let me check if existing tests suffice.